doc/en/development_guide.rst

```diff
@@ -0,0 +1,151 @@
+
+
+Code Style
+----------
+
+* `PEP-8 <https://www.python.org/dev/peps/pep-0008>`_
+* `flake8 <https://pypi.python.org/pypi/flake8>`_ for quality checks
+* `invoke <http://www.pyinvoke.org/>`_ to automate development tasks
+
+
+Branches
+--------
+
+Currently we have two long term branches:
+
+* ``master``: contains the code for the next bugfix release.
+* ``features``: contains the code with new features for the next minor release.
+
+The official repository usually does not contain topic branches, developers and contributors should create topic
+branches in their own forks.
+
+Exceptions can be made for cases where more than one contributor is working on the same
+topic or where it makes sense to use some automatic capability of the main repository (such as automatic docs from
+`readthedocs <readthedocs.org>`_ for a branch dealing with documentation refactoring.
+
+Issues
+------
+
+Any question, feature, bug or proposal is welcome as an issue. Users are encouraged to use them whenever they need.
+
+GitHub issues should use labels to categorize them. Labels should be created sporadically, to fill a niche; we should
+avoid creating labels just for the sake of creating them.
+
+Here is a list of labels and a brief description mentioning their intent.
+
+
+**Type**
+
+* ``bug``: problem that needs to be addressed.
+* ``backward compatibility``: issue that will cause problems with old pytest versions.
+* ``deprecation``: feature that will be deprecated in the future.
+* ``docs``: documentation missing or needing clarification.
+* ``enhancement``: new feature or API change, should be merged into ``features``.
+* ``feature-branch``: new feature or API change, should be merged into ``features``.
+* ``infrastructure``: improvement to development/release/CI structure.
+* ``performance``: performance or memory problem/improvement.
+* ``proposal``: proposal for a new feature, often to gather opinions or design the API around the new feature.
+* ``question``: question regarding usage, installation, internals or how to test something.
+* ``refactoring``: internal improvements to the code.
+* ``regression``: indicates a problem that was introduced in a release which was working previously.
+
+**Status**
+
+* ``critical``: grave problem or usability issue that affects lots of users.
+* ``easy``: easy issue that is friendly to new contributors.
+* ``help wanted``: core developers need help from experts on this topic.
+* ``needs information``: reporter needs to provide more information; can be closed after 2 or more weeks of inactivity.
+
+
+**Plugin (internal or external)**
+
+* ``junitxml``
+* ``cache-plugin``
+* ``capture``
+* ``collection``
+* ``doctests-plugin``
+* ``fixtures``
+* ``nose``
+* ``parametrize``
+* ``pytester-plugin``
+* ``selection``
+* ``reporting``
+* ``tracebacks``
+* ``unittest-plugin``
+* ``vendored-pluggy``
+* ``warnings-plugin``
+* ``xdist``
+
+
+**OS**
+
+Issues specific to an operating systems.
+
+* ``windows``:
```

Comment:
I propose to add `linux` and `mac` to the list.

Reply:
all plugin ones should be just like `plugin: name`